### PR TITLE
CB-8226 'cordova platform add' : Look up version in config.xml if no version specified

### PR DESCRIPTION
**Pull request by omefire**
CB-8226 
'cordova platform add' : Look up version in config.xml if no version specified

### CODE REVIEW

**Comment by omefire** [comment]
These failures seem unrelated to my changes. Ran the tests again on an Ubuntu box and all tests pass. I think these failures are intermittent and sometimes they show up, sometimes they don't.

**Comment by omefire** [comment]
Hey @agrieve, I've just addressed your review comments. could you please take a look again ? Thanks.

File: cordova-lib/src/cordova/platform.js [hunk 1]
```diff
@@ -78,62 +76,119 @@ function add(hooksRunner, projectRoot, targets, opts) {
...
+	            if (!(platform in platforms)) {
+	                return getPlatformDetailsFromDir(target);
+	            }
+	            else {
+	                if (!version) {
+	                    events.emit('verbose', 'No version supplied. Retrieving version from config.xml...');
+	                }
+	                version = version || getVersionFromConfigFile(platform, cfg);
+	                if (version) {
+	                    version = version.replace(/^file:\/\//, ''); // If version is a folder, remove the 'file://' prefix from it.
```
Comment (agrieve) [inline cordova-lib/src/cordova/platform.js]: Couldn't figure out how to reply to old comment, so creating a new one again :S.

If it starts with "file:", then it's a URL (with file: being the scheme). If it has any `:`s, or spaces, then it's a malformed URL that most parses allow. However, we really should support properly formatted URLs by using the url.parse method.

**Comment by agrieve** [comment]
Looks good except for the URL thing :)

**Comment by omefire** [comment]
Using url.parse introduces additional complexity. Especially, Windows paths are not going to be handled properly: the drive letter gets omitted after parsing. This leads to more confusion for the user.
Linux/Unix absolute paths would be handled appropriately, but Windows's wouldn't.

I suggest we remove the file:// prefix and not make config.xml contain urls for folder paths.
Take the following examples:
1-  url.parse('file://D:/Cordova/cordova-lib') => 
{ 
  protocol: 'file:',
  slashes: true,
  auth: null,
  host: 'd',
  port: null,
  hostname: 'd',
  hash: null,
  search: null,
  query: null,
  pathname: '/Cordova/cordova-lib',
  path: '/Cordova/cordova-lib',
  href: 'file://d/Cordova/cordova-lib' 
}

2- url.parse('file://D:\Cordova\cordova-lib') => 
{ 
  protocol: 'file:',
  slashes: true,
  auth: null,
  host: 'd',
  port: null,
  hostname: 'd',
  hash: null,
  search: null,
  query: null,
  pathname: '/:Cordovacordova-lib',
  path: '/:Cordovacordova-lib',
  href: 'file://d/:Cordovacordova-lib' 
}

3- url.parse('file:///Cordova/cordova-lib') =>
{ 
  protocol: 'file:',
  slashes: true,
  auth: null,
  host: '',
  port: null,
  hostname: '',
  hash: null,
  search: null,
  query: null,
  pathname: '/Cordova/cordova-lib',
  path: '/Cordova/cordova-lib',
  href: 'file:///Cordova/cordova-lib' 
}

So, my suggestion is to only support plain filesystem paths for folders in config.xml.
What do you think ?

**Comment by agrieve** [comment]
I agree that's some unnecessary complexity. Let's just drop supporting of file: prefix.

**Comment by omefire** [comment]
Done! 'file://' prefix support dropped. :)

**Comment by agrieve** [comment]
Looks good I think. Just tried to pull it in and there are many merge conflicts unfortunately. Would you be able to rebase?

**Comment by omefire** [comment]
Working on it ...

**Comment by omefire** [comment]
travis-ci failure not related to my changes.

**Comment by omefire** [comment]
@agrieve, have you had a chance to take another look at this ?   --Thanks.

**Comment by agrieve** [comment]
Merged! However - after a closer look at the unit tests, I highly encourage you to try and refactor things so that the tests do not mock out private methods within modules (e.g. like `resolvePath()`). Tests that do that make it very hard to change the code going forward, as it tests implementation details rather than function pre/post-conditions.

**Comment by omefire** [comment]
@agrieve , I will keep that in mind. Thanks for taking the time to review this !

**Pull request closed without merging**